src/adapter/src/session.rs

```diff
@@ -1276,6 +1279,8 @@ pub enum TransactionOps<T> {
         determination: TimestampDetermination<T>,
         /// The cluster used to execute peeks.
         cluster_id: ClusterId,
+        /// Whether this peek needs to be linearized.
+        requires_linearization: bool,
```

Comment:
slight preference to making this an enum
```
enum RequireLinearization {
  Required,
  NotRequired,
}
```

Reply:
Done